Problems/1874/ans_1874_YH.py

```diff
@@ -0,0 +1,53 @@
+#https://www.acmicpc.net/problem/1874
+
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+class Stack:
+    def __init__(self):
+        self.head = None
+    def push(self,data):
+        new_node = Node(data)
+        new_node.next = self.head
+        self.head = new_node
+    def pop(self):
+        if self.is_empty():
+            return -1
+        data = self.head.data
+        self.head = self.head.next
+        return data
+    def is_empty(self):
+        if self.head:
+            return False
+        return True
+    def peak(self):
+        if self.is_empty():
+            return -1
+        return self.head.data
+
+def solve():
+    N = int(input())
+    cnt=1
+    stack = Stack()
+    output = []
+    for _ in range(N):
+        num = int(input())
+        while cnt <= num:
+            stack.push(cnt)
+            output.append("+")
+            cnt+=1
+        
+        if stack.peak() == num:
+            stack.pop()
```

Comment:
```suggestion
        if stack.pop() == num:
```
그냥 pop 치면서 해도 될거같아보이네